de.metas.adempiere.adempiere/client/src/main/java-legacy/org/compiere/apps/AEnv.java

```diff
@@ -633,30 +633,33 @@ public static void zoom(
 		{
 			return;
 		}
-		
+
 		zoom(RecordZoomWindowFinder.newInstance(TableName, Record_ID)
 				.setSO_Window_ID(AD_Window_ID)
 				.setPO_Window_ID(PO_Window_ID));
 	}
-	
+
 	private static final void zoom(final RecordZoomWindowFinder zoomInfo)
 	{
 		final int windowIdToUse = zoomInfo.findAD_Window_ID();
-		if(windowIdToUse <= 0)
+		if (windowIdToUse <= 0)
 		{
 			log.warn("No AD_Window_ID found to zoom for {}", zoomInfo);
 			return;
 		}
-		
+
 		final MQuery query = zoomInfo.createZoomQuery();
-		
-		final AWindow frame = new AWindow();
-		if (!frame.initWindow(windowIdToUse, query))
-		{
-			return;
-		}
-		addToWindowManager(frame);
-		showCenterScreenOrMaximized(frame);
+
+		// task #797 Make sure the windows are displayed in the same thread as the process
```

Comment:
hi, i don't fully understand the comment..from the invokelater-javadoc i understand that that with this method, we actually make sure the windows are displayed in the *AWT event dispatching thread*, even although this zoom invokaction might or might be coming from another thread..
am i getting it right?